firebase-ai/src/main/kotlin/com/google/firebase/ai/FirebaseAI.kt

```diff
@@ -212,15 +216,31 @@ internal constructor(
      * Returns the [FirebaseAI] instance for the provided [FirebaseApp] and [backend].
      *
      * @param backend the backend reference to make generative AI requests to.
+     * @param useLimitedUseAppCheckTokens when sending tokens to the backend, this option enables
+     * the usage of App Check's limited-use tokens instead of the standard cached tokens.
+     *
+     * A new limited-use tokens will be generated for each request; providing a smaller attack
+     * surface for malicious parties to hijack tokens. When used alongside replay protection,
+     * limited-use tokens are also _consumed_ after each request, ensuring they can't be used again.
+     *
+     * _This flag is set to `false` by default._
+     *
+     * **Important:** Replay protection is not currently supported for the FirebaseAI backend. While
+     * this feature is being developed, you can still migrate to using limited-use tokens. Because
+     * limited-use tokens are backwards compatible, you can still use them without replay
+     * protection. Due to their shorter TTL over standard App Check tokens, they still provide a
+     * security benefit. Migrating to limited-use tokens sooner minimizes disruption when support
+     * for replay protection is added.
      */
     @JvmStatic
     @JvmOverloads
     public fun getInstance(
       app: FirebaseApp = Firebase.app,
-      backend: GenerativeBackend
+      backend: GenerativeBackend,
+      useLimitedUseAppCheckTokens: Boolean = false,
```

Comment:
You need to add a new, handcrafted overload for this due to https://lankydan.dev/backwards-compatibility-of-kotlins-default-arguments